Call to Arms UII

Flam3sDOTA opened the pull request:
Added Call to Arms Window Popup,
Replaced old Discord Link that exists on the loading screen

Flam3sDOTA committed:
Call to Arms UII

Added Call to Arms Window Popup,
Replaced old Discord Link that exists on the loading screen

n-gao commented:
Please add a cookie to avoid showing the popup every time I start a new game. I just want to see this once per update note. So best store the version of the popup and update it once you change the text.

n-gao commented on content/dota_addons/legion_td/panorama/layout/custom_game/main_hud.xml:
```diff
@@ -4,13 +4,30 @@
+					<Label id="Message" text="#Call_To_Arms_Description"/>
+				</Panel>
+				<Label id="VersionNumber" text="#Call_to_Arms"/>
+				<Button id="DiscordButton"  onactivate="ExternalBrowserGoToURL('https://discord.gg/pmMnfdA');" onmouseover="UIShowTextTooltip( Join the community!)"  onmouseout="UIHideTextTooltip()">
```
The onmouseover text is not localized

Flam3sDOTA committed:
Update main_hud.xml

Flam3sDOTA committed:
Update main_hud.xml

Flam3sDOTA committed:
Added Button and Fixed Round UI

Added Button to Toggle Spring Cleaning UI
Fixed Round UI off-scaling
Changed Loading Screen Background to Tease the New Builder

Flam3sDOTA committed:
Fixed Enchantress

n-gao commented on game/dota_addons/legion_td/scripts/vscripts/abilities/modifier_enchantress_untouchable_lua_debuff.lua:
```diff
@@ -0,0 +1,113 @@
+modifier_enchantress_untouchable_lua_debuff = class({})
```
This file should probably be in `vscripts/abilities/waveunits/` like `enchantress_untouchable_lua.lua`.

n-gao commented on game/dota_addons/legion_td/scripts/vscripts/abilities/modifier_enchantress_untouchable_lua.lua:
```diff
@@ -0,0 +1,55 @@
+modifier_enchantress_untouchable_lua = class({})
```
This file should probably be in `vscripts/abilities/waveunits/` like `enchantress_untouchable_lua.lua`.

n-gao commented on content/dota_addons/legion_td/panorama/layout/custom_game/main_hud.xml:
```diff
@@ -4,13 +4,36 @@
     </scripts>
     <Panel hittest="false" class="Hud">
         <Panel hittest="false" class="DontKnow">
+		
+			<Panel hittest="false" id="PopupWindow">
+				<Panel id="BackgroundImage"/>
+				<Panel id="Contents">
+					<Label id="Title" text="#SPRING_CLEANING"/>
+					<Label id="Message" text="#Call_To_Arms_Description"/>
+				</Panel>
+				<Label id="VersionNumber" text="#Call_to_Arms"/>
+				<Button id="DiscordButton"  onactivate="ExternalBrowserGoToURL('https://discord.gg/pmMnfdA');" onmouseover="UIShowTextTooltip(#Ad_updated)"  onmouseout="UIHideTextTooltip()">
+					<Image id="DiscordImage" src="file://{images}/custom_game/game_info/Discord.png" />
+				</Button>
+				<Panel id="Buttons" class="PopupButtonRow">
+					<TextButton class="ButtonBevel" text="#DOTA_Continue" onactivate="OnContinueButtonPressed()" />
+				</Panel>
+			</Panel>
+	
+			<Panel id="StyleButtons" hittest="false">
+				<Button class="ButtonBevel" id="NextButton" onactivate="OnCleaningButtonPressed()" onload="DOTAShowTextTooltip('#Announcement')" onmouseover="UIShowTextTooltip(#Announcement)"  onmouseout="UIHideTextTooltip()">
+					<Label id="Style2ButtonLabel" text="#SPRING_CLEANING"/>
+				</Button>
+			</Panel>
```
This still shows up every game right? Could you move it to the loading screen instead? So it a) isn't popping up every game and b) isn't happens earlier in the game

Flam3sDOTA replied:
It is hidden at the start, gets open  when someone clicks to a button.  (the button blinks at the start of the game for 10 times, like the loading screen discord image)

![unknown](https://user-images.githubusercontent.com/30335579/72209284-b4806c80-34bd-11ea-9845-cf6f45bf0027.png)

There is no possible place on loading screen and if I put it as extra, it will cover whole screen and will include many text along with Tips etc.

a-) Either I remove the how to play & Tips screen and replace it with news like how Auto Chess Does

b-) It stays where it is at the moment.

Flam3sDOTA committed:
Relocated Modifiers

Relocated Modifiers
